### ISSUE TITLE
Forms: Form edits cannot be saved

### ISSUE THREAD

**Issue by thomasddn** [umbraco/Umbraco.Forms.Issues#393]
Reproduction
------------

### Bug summary

Form edits cannot be saved. When looking in the logs, the following error appears:

```
System.NullReferenceException:
   at Umbraco.Forms.Web.Editors.FormController+<>c__DisplayClass17_0.<ContainsSensitiveData>b__0 (Umbraco.Forms.Web, Version=8.5.1.0, Culture=neutral, PublicKeyToken=null)
   at System.Collections.Generic.List`1.ForEach (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Umbraco.Forms.Web.Editors.FormController.ContainsSensitiveData (Umbraco.Forms.Web, Version=8.5.1.0, Culture=neutral, PublicKeyToken=null)
   at Umbraco.Forms.Web.Editors.FormController.SaveForm (Umbraco.Forms.Web, Version=8.5.1.0, Culture=neutral, PublicKeyToken=null)
   at lambda_method (Anonymously Hosted DynamicMethods Assembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor+ActionExecutor+<>c__DisplayClass6_2.<GetExecutor>b__2 (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor+ActionExecutor.Execute (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Web.Http.Controllers.ApiControllerActionInvoker+<InvokeActionAsyncCore>d__1.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Web.Http.Filters.ActionFilterAttribute+<CallOnActionExecutedAsync>d__6.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Web.Http.Filters.ActionFilterAttribute+<CallOnActionExecutedAsync>d__6.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)

   ...

   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Web.Http.Filters.AuthorizationFilterAttribute+<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Web.Http.Filters.AuthorizationFilterAttribute+<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Web.Http.Controllers.ExceptionFilterResult+<ExecuteAsync>d__6.MoveNext (System.Web.Http, Version=5.2.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
```

### Specifics

Umbraco: v8.5.5
Umbraco Forms: v8.5.1
Umbraco Forms config - StoreUmbracoFormsInDb: True

### Steps to reproduce

1. Sign in as a user added to the Administrator group, but not to the "Sensitive data" group.
2. Create a new form, based on the "Empty form".
3. Give the form a name and click Save. The Consent question is already added by default.
4. Add a new question, answer type "short answer".
5. Click on Save. Save does not work, an X (instead of Save) is shown in the green button for a few seconds.

Going away from the form editor and coming back, yields the same unexpected result.

### Expected result

The changes to the form should be saved. Or at least a message should be shown to indicate that the current user has insufficient security rights.

### Actual result

The changes to the form are not saved. It doesn't have to be on a new form, the same issue appears on existing forms.

---
_This item has been added to our backlog [AB#7907](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/7907)_

---
_This item has been added to our backlog [AB#7927](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/7927)_

**Comment by callumbwhyte** [comment]
Can confirm this is happening for me too - users need to be in the "Sensitive Data" user group in order to save forms. Occurred after upgrading to Forms 8.5+

Seems like a pretty severe bug, as by default only the Admin user is in this group, so hopefully we can get a speedy fix!

**Comment by nul800sebastiaan** [comment]
Oh my! We'll have to have a look and fix soon indeed!

**Comment by nul800sebastiaan** [comment]
FYI: Can reproduce the error and can confirm it's a regression, this used to work in Forms 8.4. 

We've recently hidden Form 8.5.x from NuGet as we also had some problems with upgrades. We'll fix this before a new Forms release comes out.

**Comment by nul800sebastiaan** [comment]
This has been fixed for 8.5.2.
